src/utils/logging.rs

```diff
@@ -1,30 +1,66 @@
-use std::io;
-use std::io::Write;
-use std::mem;
-use std::sync::atomic::{AtomicUsize, Ordering};
-use std::sync::{Arc, Weak};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Weak,
+};
 
 use chrono::Local;
 use console::{style, Color};
 use indicatif::ProgressBar;
 use lazy_static::lazy_static;
+use log::max_level;
 use parking_lot::RwLock;
 
+// Quiet mode decides whether to print something based on `--quiet` flag.
+lazy_static! {
+    static ref QUIET_MODE: AtomicBool = AtomicBool::new(false);
+}
+
+pub fn quiet_mode() -> bool {
+    QUIET_MODE.load(Ordering::Relaxed)
+}
+
+pub fn set_quiet_mode(yes: bool) {
+    QUIET_MODE.store(yes, Ordering::Relaxed);
+}
+
+// NOTE: Remove `allow`s after first use.
+#[allow(unused_macros)]
+macro_rules! quiet_println {
+    ($($tt:tt)*) => {{
+        if !crate::utils::logging::quiet_mode() {
+            println!($($tt)*);
```

Comment:
is this macro to prevent allocations? If it is - super clever!!

Reply:
I don't believe it prevents it, as quiet_mode is decided during runtime.
It's so that I can put `quiet_println!` in place of `println!` in most places. (it will be runner up PR)

Reply:
Cool, sounds good!